cbmc/proofs/mqtt_state.c

```diff
@@ -545,43 +568,41 @@ void free_IotMqttSubscriptionList( IotListDouble_t *pSub )
  * IotMqttPublishInfo
  ****************************************************************/
 
+void bound_IotMqttPublishInfo( IotMqttPublishInfo_t *pInfo )
+{
+    #ifdef TOPIC_NAME_LENGTH_MAX
+        __CPROVER_assume( pInfo->topicNameLength < TOPIC_NAME_LENGTH_MAX &&
+                          VALID_CBMC_SIZE( pInfo->topicNameLength ) );
+    #else
+      __CPROVER_assume( pInfo->topicNameLength <= UINT16_MAX );
+    #endif
+
+    #ifdef PAYLOAD_LENGTH_MAX
+        __CPROVER_assume( pInfo->payloadLength < PAYLOAD_LENGTH_MAX &&
+                          VALID_CBMC_SIZE( pInfo->payloadLength ) );
+    #else
+      __CPROVER_assume( pInfo->payloadLength <= UINT16_MAX );
+    #endif
+}
+
 IotMqttPublishInfo_t *allocate_IotMqttPublishInfo( IotMqttPublishInfo_t *pInfo )
 {
   if ( pInfo == NULL ) pInfo = malloc_can_fail( sizeof( *pInfo ) );
   if ( pInfo == NULL ) return NULL;
 
+  bound_IotMqttPublishInfo( pInfo );
```

Comment:
Again, is this just a refactoring?  It is the same thing.

Reply:
See previous comments.

Reply:
OK